Utilities/File.cpp

```diff
@@ -1375,6 +1375,17 @@ const std::string& fs::get_config_dir()
 		std::replace(dir.begin(), dir.end(), '\\', '/');
 
 		dir.resize(dir.rfind('/') + 1);
+#elif __APPLE__
```

Comment:
The entirety of this elif doesn't need to be cloned, you could just check for both MacOS and Linux and then only #ifdef the
```cpp
dir = home + "/Library/Application Support"s;
```
and 
```cpp
if (const char* home = ::getenv("XDG_CONFIG_HOME"))
			dir = home;
 		else if (const char* home = ::getenv("HOME"))
 			dir = home + "/.config"s;
```

Reply:
I am not too fond of nested ifdefs, but perhaps you are right that it is better in this case. Changed.